docs/proposals/2841-flux-hpc/README.md

```diff
@@ -0,0 +1,275 @@
+
+### API Design
+
+The proposed changes will integrate into the existing `v1alpha1` API structures. We will add a new field, `hpc`, to the `MLPolicySource` struct. This aligns Flux with other runtimes like Torch and MPI.
+
+#### 1. `ClusterTrainingRuntime` and `TrainingRuntime`
+
+The proposal will leverage the existing `ClusterTrainingRuntime` (cluster-scoped) and `TrainingRuntime` (namespace-scoped) CRDs. These objects serve as reusable templates, and no changes are needed for their top-level structure.
+
+#### 2. Enhancing `MLPolicySource`
+
+We will add the `HPC` field to the `MLPolicySource` struct, which is embedded within the `MLPolicy`.
+
+```go
+// MLPolicySource is the runtime-specific configuration
+// One of the following specs can be set.
+type MLPolicySource struct {
+    // torch defines the configuration for the PyTorch runtime.
+    // +optional
+	Torch *TorchMLPolicySource `json:"torch,omitempty"`
+
+	// mpi defines the configuration for the MPI Runtime.
+	// +optional
+	MPI *MPIMLPolicySource `json:"mpi,omitempty"`
+
+	// hpc defines the configuration for an hpc runtime (e.g., Flux Framework).
+	// This is the new field being added.
+	// +optional
+	HPC *HPCMLPolicySource `json:"hpc,omitempty"`
+}
+
+// HPCMLPolicySource represents an HPC runtime configuration.
+type HPCMLPolicySource struct {
+	// Manager specifies the workload manager to use.
+	// +kubebuilder:default="flux"
+	// +optional
+	Manager string `json:"manager,omitempty"`
+
+	// Tasks is the number of tasks to provide to the workload manager.
+	// +optional
+	Tasks *int32 `json:"tasks,omitempty"`
+
+	// Settings provides a key-value map for manager-specific configurations.
+	// +optional
+	Settings map[string]string `json:"settings,omitempty"`
+}
+```
+
+**Example `ClusterTrainingRuntime`:**
+
+This is an example created by an administrator.
+
+```yaml
+apiVersion: trainer.kubeflow.org/v1alpha1
+kind: ClusterTrainingRuntime
+metadata:
+  name: hpc-flux-runtime
+  labels:
+    trainer.kubeflow.org/framework: flux
+spec:
+  # The hpc policy is correctly nested under mlPolicy
+  mlPolicy:
+    hpc:
+      manager: flux
+      settings:
+        flux-view-image: "ghcr.io/converged-computing/flux-view-ubuntu:tag-jammy"
+        flux-network-device: "eth0"
+        flux-queue-policy: "fcfs"
+
+  # The base JobSet template that the Flux plugin will start with
+  template:
+    spec:
+      replicatedJobs:
+        - name: trainer
+          template:
+            spec:
+              completionMode: Indexed
+              completions: 1
+              parallelism: 1
+              template:
+                spec:
+                  containers:
+                    - name: node
+                      image: "placeholder-image"
+```
+
+**Example Consuming `TrainJob` YAML:**
+
+```yaml
+apiVersion: trainer.kubeflow.org/v1alpha1
+kind: TrainJob
+metadata:
+  name: lammps-flux-interactive
+spec:
+  # Reference the pre-defined runtime by name
+  runtime: hpc-flux-runtime
```

Comment:
This should have correct spec, check example here: https://www.kubeflow.org/docs/components/trainer/operator-guides/runtime/#example-of-clustertrainingruntime

Reply:
I appreciate this! I was having a really hard time figuring out what was right.